### DISCUSSION TITLE
Does Visual Studio Code extension work for Vue2?

### DISCUSSION THREAD

**Discussion by Darkzarich** [vuejs/language-tools#4825] [Q&A]
Hello!

Extension from this repo work great for Vue 3 but it kind of errors out for Vue 2 (`"vue": "^2.7.14"`) in some places. I think it generally work, but then I see these typing warnings though I do not use TypeScript in this old project:

![image](https://github.com/user-attachments/assets/e3ef03bf-404a-4ed7-9a01-2eeb0575fe76)

I don't think I can properly type it in Vue 2.

Maybe I didn't setup something? or should I use a different extension for Vue 2?

**Comment by RayGuo-ergou** [comment]
Do you have `checkJs: true` in your `jsconfig.json`? If yes try to remove that, if not try to explicitly set to false.

**Reply by Darkzarich** [reply]
Hello! Yes, I do have that set as `true`. 

But if I set it to `false` will it still suggest me fields from `data` and `props`? 

I also use it for `*.js` files but for that matter I could just add all `*.vue` files to `exclude`

**Reply by RayGuo-ergou** [reply]
It just tell typescript to not check errors for javascript files, all other functionality should work: https://www.typescriptlang.org/tsconfig/#checkJs

**Reply by Darkzarich** [reply]
I understand what it does, it does get rid of errors but it also gets rid of any type checking :(

I tried adding this to my `jsconfig.json` :

```
"exclude": ["node_modules", "dist", "*/**/*.vue"],
```

so I at least have type check in my js files but it's not ideal unfortunately.

As for as I remember Vue 2 had terrible TypeScript support so maybe it can't be helped.

**Reply by RayGuo-ergou** [reply]
Ah i see, yes you could do that or 

1. use `@vue-ignore` or `@ts-ignore`
2. Use jsdoc, e.g. declare nativeType as the button type union
3. Probably is not really needed but can have another `jsconfig.json` for only vue files.
